pkg/client/apiutil/restmapper_test.go

```diff
@@ -210,6 +210,28 @@ func TestLazyRestMapperProvider(t *testing.T) {
 		g.Expect(crt.GetRequestCount()).To(gmg.Equal(4))
 	})
 
+	t.Run("LazyRESTMapper KindsFor should work correctly with empty versions list", func(t *testing.T) {
+		g := gmg.NewWithT(t)
+
+		httpClient, err := rest.HTTPClientFor(restCfg)
+		g.Expect(err).NotTo(gmg.HaveOccurred())
+
+		crt := newCountingRoundTripper(httpClient.Transport)
+		httpClient.Transport = crt
+
+		lazyRestMapper, err := apiutil.NewDynamicRESTMapper(restCfg, httpClient)
+		g.Expect(err).NotTo(gmg.HaveOccurred())
+
+		g.Expect(crt.GetRequestCount()).To(gmg.Equal(0))
+
+		kinds, err := lazyRestMapper.KindsFor(schema.GroupVersionResource{Group: "autoscaling", Resource: "horizontalpodautoscaler"})
+		g.Expect(err).NotTo(gmg.HaveOccurred())
+		g.Expect(len(kinds)).To(gmg.Equal(2))
+		g.Expect(kinds[0].Kind).To(gmg.Equal("HorizontalPodAutoscaler"))
+		g.Expect(kinds[1].Kind).To(gmg.Equal("HorizontalPodAutoscaler"))
+		g.Expect(crt.GetRequestCount()).To(gmg.Equal(4))
+	})
```

Comment:
Maybe you could add no-version tests for all functions in the interface? Like https://github.com/kubernetes-sigs/controller-runtime/blob/73949d01fbbebeb5e2ce20f7e3b014832237186f/pkg/client/apiutil/restmapper_test.go#L88-L138

Reply:
Good idea, done!

Reply:
However, checked for non-empty lists rather than length of 1